src/web-auth/captcha.js

```diff
@@ -0,0 +1,117 @@
+// eslint-disable-next-line no-unused-vars
+import Authentication from '../authentication';
+import object from '../helper/object';
+
+var defaults = {
+  lang: 'en',
+  templates: {
+    'auth0': function (challenge) {
+      var message = challenge.type === 'code' ?
+        'Enter the code shown above' :
+        'Solve the formula shown above';
+      return '<div class="captcha-challenge">\n' +
+        '  <img src="' + challenge.image + '" />\n' +
+        '  <button type="button" class="captcha-reload">↺</button>\n' +
+        '</div>\n' +
+        '<input type="text" name="captcha"\n' +
+        '  class="form-control captcha-control"\n' +
+        '  placeholder="' + message + '" />';
+    }
+    ,
+    'recaptcha_v2': function () {
+      return '<div class= "recaptcha" ></div><input type="hidden" name="captcha" />';
+    }
+  }
+};
+
+function handleAuth0Provider(element, options, challenge, load) {
+  element.innerHTML = options.templates[challenge.provider](challenge);
+  element.querySelector('.captcha-reload').addEventListener('click', function (e) {
+    e.preventDefault();
+    load();
+  });
+}
+
+function injectRecaptchaScript(element, lang, callback) {
+  var callbackName = 'recatpchaCallback_' + Math.floor(Math.random() * 1000001);
+  window[callbackName] = function () {
+    delete window[callbackName];
+    callback();
+  };
+  var script = window.document.createElement('script');
+  script.src = 'https://www.google.com/recaptcha/api.js?hl=' + lang + '&onload=' + callbackName;
+  script.async = true;
+  window.document.body.appendChild(script);
+}
+
+function handleRecaptchaProvider(element, options, challenge) {
+  var widgetId = element.hasAttribute('data-wid') && element.getAttribute('data-wid');
+
+  function setValue(value) {
+    var input = element.querySelector('input[name="captcha"]');
+    input.value = value || '';
+  }
+
+  if (widgetId) {
+    setValue();
+    window.grecaptcha.reset(widgetId);
+    return;
+  }
+
+  element.innerHTML = options.templates[challenge.provider](challenge);
+
+  var recaptchaDiv = element.querySelector('.recaptcha');
+
+  injectRecaptchaScript(element, options.lang, function () {
+    widgetId = window.grecaptcha.render(recaptchaDiv, {
+      callback: setValue,
+      'expired-callback': function () { setValue(); },
+      'error-callback': function () { setValue(); },
+      sitekey: challenge.siteKey
```

Comment:
@jfromaniello is this change in the casing an oversight or intended?

```suggestion
      siteKey: challenge.siteKey
```

Reply:
the auth0 api returns `siteKey` (camel-case), the google recaptcha api expects `sitekey` (all lower case)